src/main/common/filter.c

```diff
@@ -99,23 +121,46 @@ float pt2FilterGainFromDelay(float delay, float dT)
     return dT / (dT + delay * CUTOFF_CORRECTION_PT2);
 }
 
+void pt2FilterInitArray(void *filter, float k_value, int count)
+{
+    pt2Filter_t *f = (pt2Filter_t*)filter;
+    f->k = k_value;
+
+    // splitting the state as below suppresses compiler warnings
+    float *state0 = &f->state[0][0];
+    float *state1 = &f->state[1][0];
+    for (int i = 0; i < count; i++) {
+        state0[i] = 0.0f;
+        state1[i] = 0.0f;
```

Comment:
(just for reference)
```suggestion
        for(int j = 0; j < 2; i++)
            f->stat[j][i] = .0f;
```